base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py

```diff
@@ -228,4 +230,11 @@ def self_valid():
         runner = XMLTestRunner(output=os.path.join(logs_dir, filename), stream=sys.stdout)
         return_values.append(runner.run(suite).wasSuccessful())
 
+    # for the opi_info.xml check
+    print("Testing '{}'".format(os.path.join(root_dir, "opi_info.xml")))
+    suite = unittest.TestSuite()
+    suite.addTests([TestOpiInfo(test) for test in loader.getTestCaseNames(TestOpiInfo)])
```

Comment:
You only need to explicitly iterate over test case names if you are creating tests dynamically (e.g. running a test suite for each opi). As your test suite currently covers all OPIs at once I think you can just do:

```suggestion
    suite.addTests(loader.loadTestsFromTestCase(TestOpiInfo))
```